### PR TITLE
Refactor Integration modules.

### PR DESCRIPTION
**Pull request by vishalpandeyvip**

### CODE REVIEW

File: apps/bamboohr/views.py [hunk 1]
```diff
@@ -193,7 +149,7 @@ def get(self, request, *args, **kwargs):
         except BambooHrConfiguration.DoesNotExist:
             return Response(
                 data={'message': 'BambooHr Configuration does not exist for this Workspace'},
-                status=status.HTTP_400_BAD_REQUEST
```
Comment (NileshPant1999) [inline apps/bamboohr/views.py:196]: can you check if client is not consuming this code.

Comment (vishalpandeyvip) [reply apps/bamboohr/views.py:196]: A method is available in bamboohr service that makes get request to this API. 
though this method is not used anywhere in fyle-integrations-settings-app.
[Link](https://github.com/fylein/fyle-integrations-settings-app/blob/master/src/app/core/services/bamboo-hr/bamboo-hr.service.ts#L35)

File: apps/bamboohr/actions.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    bamboo_connection_1 = next(connection for connection in connections if connection['name'] == BAMBOO_HR['connections'][0])
+    bamboo_connection_2 = next(connection for connection in connections if connection['name'] == BAMBOO_HR['connections'][1])
+
+    configuration.recipe_status = False
+    configuration.save()
+
+    connection = connector.connections.post(
+        managed_user_id=org.managed_user_id,
+        connection_id=bamboo_connection_1['id'],
+
```
Comment (NileshPant1999) [inline apps/bamboohr/actions.py:27]: remove this extra line

File: apps/bamboohr/actions.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+from apps.bamboohr.models import BambooHr, BambooHrConfiguration
+from apps.names import BAMBOO_HR
+
+@handle_workato_exception(task_name = 'Disconnect BambooHR')
+def disconnect_bamboohr(org_id, configuration:BambooHrConfiguration, bamboohr: BambooHr):
+    org = Org.objects.get(id=org_id)
+
+    connector = Workato()
+    connections = connector.connections.get(managed_user_id=org.managed_user_id)['result']
+    bamboo_connection_1 = next(connection for connection in connections if connection['name'] == BAMBOO_HR['connections'][0])
```
Comment (NileshPant1999) [inline apps/bamboohr/actions.py:18]: can you rename this to connection type i mean one is a scheduled job other one is a sync

Comment (NileshPant1999) [reply apps/bamboohr/actions.py:18]: connection1 and connection2 is very generic

File: apps/bamboohr/views.py [hunk 4]
```diff
@@ -37,7 +29,7 @@ def get(self, request, *args, **kwargs):
         except BambooHr.DoesNotExist:
             return Response(
                 data={'message': 'Bamboo HR Details Not Found'},
-                status=status.HTTP_400_BAD_REQUEST
+                status=status.HTTP_404_NOT_FOUND
```
Comment (NileshPant1999) [inline apps/bamboohr/views.py:32]: again check in the client if there is a comparison to the status code

File: apps/travelperk/views.py [hunk 5]
```diff
@@ -172,7 +133,7 @@ def post(self, request, *args, **kwargs):
             )
     
         elif 'authorization_status' in connection:
```
Comment (NileshPant1999) [inline apps/travelperk/views.py:135]: authorization will always be present in the connection dict, we can just return response outside of if

File: apps/travelperk/views.py [hunk 6]
```diff
@@ -195,7 +156,7 @@ def get(self, request, *args, **kwargs):
         except TravelPerkConfiguration.DoesNotExist:
             return Response(
                 data={'message': 'Configuration does not exist for this Workspace'},
-                status=status.HTTP_400_BAD_REQUEST
+                status=status.HTTP_404_NOT_FOUND
```
Comment (NileshPant1999) [inline apps/travelperk/views.py:159]: check with client if there is any comparison done with status code.

File: apps/travelperk/views.py [hunk 7]
```diff
@@ -234,40 +195,13 @@ class TravelperkConnection(generics.ListCreateAPIView):
...
-                status=status.HTTP_400_BAD_REQUEST
-            )
+        # case of an error Response
+        if isinstance(connection_id, Response):
+            return connection_id
+        
+        return Response(
+            data={'message': {'connection_id': connection_id}},
+            status=status.HTTP_200_OK
+        )
```
Comment (NileshPant1999) [inline apps/travelperk/views.py:207]: new line

**Review by NileshPant1999 — APPROVED** [review]

**Pull request merged**